.werft/jobs/build/validate-changes.ts

```diff
@@ -1,12 +1,38 @@
 import { exec } from "../../util/shell";
 import { Werft } from "../../util/werft";
+import { JobConfig } from "./job-config";
 
-export async function validateChanges(werft: Werft) {
+export async function validateChanges(werft: Werft, config: JobConfig) {
     werft.phase('validate-changes', 'validating changes');
     try {
-        exec(`pre-commit run --all-files --show-diff-on-failure`);
-        werft.done('validate-changes');
+        branchNameCheck(werft, config)
+        preCommitCheck(werft)
     } catch (err) {
         werft.fail('validate-changes', err);
     }
+    werft.done('validate-changes');
+}
+
+// Branch names cannot be longer than 20 characters.
+// We plan to remove this limitation once we move to previews on Harvester VMs.
+async function branchNameCheck(werft: Werft, config: JobConfig) {
+    if (!config.noPreview) {
+        const maxBranchNameLength = 20;
+        werft.log("check-branchname", `Checking if branch name is shorter than ${maxBranchNameLength} characters.`)
+
+        if (config.previewEnvironment.destname.length > maxBranchNameLength) {
+            throw new Error(`The branch name ${config.previewEnvironment.destname} is more than ${maxBranchNameLength} character. Please choose a shorter name!`)
+        }
+        werft.done("check-branchname")
+    }
+}
+
+async function preCommitCheck(werft: Werft) {
+    werft.log("pre-commit checks", "Running pre-commit hooks.")
+    try {
```

Comment:
As you're re-throwing the error I would suggest removing the try/catch logic here all-together ☺️

Reply:
Ah nice catch...

I think we could follow this pattern on other areas too:
* One public function that is directly called by build.ts
* The public function would call private functions on a try-catch block
* Private functions will throw errors instead of running `werft.fail`